prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -4225,6 +4241,63 @@ subroutine readCoordinationNumber(node, input, geo, cnDefault, cutDefault)
   end subroutine readCoordinationNumber
 
 
+#:if WITH_MBD
+
+  subroutine readDispTs(node, input)
+    type(fnode), pointer, intent(in) :: node
+    type(TMbdInit), intent(out) :: input
+
+    type(string) :: buffer
+    type(fnode), pointer :: child
+
+    ! TODO there should be a mechanism that allows either TS or MBD, not both,
+    ! or we have to initialize two different mbd_calc instances
+    input%method = 'ts'
+    call getChildValue(node, "EnergyAccuracy", input%ts_ene_acc, input%ts_ene_acc, modifier=buffer,&
+        & child=child)
+    call convertByMul(char(buffer), energyUnits, child, input%ts_ene_acc)
+    call getChildValue(node, "ForceAccuracy", input%ts_f_acc, input%ts_f_acc, modifier=buffer, child=child)
+    call convertByMul(char(buffer), forceUnits, child, input%ts_f_acc)
+    call getChildValue(node, "Damping", input%ts_d, input%ts_d)
+    call getChildValue(node, "RangeSeparation", input%ts_sr, input%ts_sr)
+    call getChildValue(node, "ReferenceSet", buffer, 'ts', child=child)
+    input%vdw_params_kind = tolower(unquote(char(buffer)))
+    call checkManyBodyDispRefName(input%vdw_params_kind, child)
+
+  end subroutine readDispTs
+
+
+  subroutine readDispMbd(node, input)
+    type(fnode), pointer, intent(in) :: node
+    type(TMbdInit), intent(out) :: input
+
+    type(string) :: buffer
+    type(fnode), pointer :: child
+
+    input%method = 'mbd-rsscs'
+    call getChildValue(node, "Beta", input%mbd_beta, input%mbd_beta)
+    call getChildValue(node, "NOmegaGrid", input%n_omega_grid, input%n_omega_grid)
+    call getChildValue(node, "KGrid", input%k_grid)
+    call getChildValue(node, "KGridShift", input%k_grid_shift, input%k_grid_shift)
+    call getChildValue(node, "ReferenceSet", buffer, 'ts', child=child)
```

Comment:
This line is causing the segfault. I have no idea, I'm sure this worked before.

Reply:
There is one bug in the initprogram.F90 preprocessor section:

--- a/prog/dftb+/lib_dftbplus/initprogram.F90
+++ b/prog/dftb+/lib_dftbplus/initprogram.F90
@@ -2048,20 +2048,20 @@ contains
           call init(dftd4, input%ctrl%dispInp%dftd4, nAtom, species0, speciesName)
         end if
         call move_alloc(dftd4, dispersion)
-      end if
-      cutOff%mCutOff = max(cutOff%mCutOff, dispersion%getRCutOff())
     #:if WITH_MBD
       elseif (allocated(input%ctrl%dispInp%mbdInp)) then
         tManyBodyDisp = .true.
         allocate (mbDispersion)
         associate (inp => input%ctrl%dispInp%mbdInp)
            inp%calculate_forces = tForces
            inp%atom_types = speciesName(species0)
            inp%coords = coord0
            if (tPeriodic) inp%lattice_vectors = latVec
            call mbDispersion%init(inp)
         end associate
      #:endif
+      end if
+      cutOff%mCutOff = max(cutOff%mCutOff, dispersion%getRCutOff())
     end if

but the second line may be in the wrong place, depending on whether MBD sets a cut-off in the same was as other dispersion models.